### DISCUSSION TITLE
Welcome to STAR Discussions!

### DISCUSSION THREAD

**Discussion by alexdobin** [alexdobin/STAR#1097] [General]
##  Welcome to STAR discussions!
**Please use Discussions if you have a question rather than an issue.**

  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together.

**Comment by IainPerry** [comment]
Hi,
I'm trying to understand observed differences between observed differentially expressed genes calculated with Deseq2 on the un-stranded counts output from STAR genecount and from the output of featurecounts gene_id.
I feel like I'm missing something obvious but in a dataset of 12,000 differentially expressed genes about 1000 different genes are not found by genecounts and 1000 are not found by featurecounts.
I suspect the issue is in splice variant calling but I was hoping for clarification if this is expected and why?
Thanks

**Comment by alexdobin** [comment]
Hi Iain,

for single-end libraries, there should not be any difference between STAR's GeneCounts and featureCounts.
The difference for paired-end libraries should be relatively small as well. The two algorithms are different for PE reads in which both mates overlap geneA, but only one of the mates overlaps geneB. STAR (following htseq-count) considers these reads "ambiguous", while featureCounts counts them towards geneA. If you make a scatter plot of the two, you should be able to see the small difference for some genes.

As to the large difference in DE genes, I think it's because you have a very large number of DE genes, 12k - that's probably more than half of all expressed genes. If look only at the top DE genes (say, ranked by p-value), the differences will hopefully be small.

Cheers
Alex

**Comment by awlButler** [comment]
Dear Alex,
I used STAR v2.7.7a to run alignReads on 2 paired-end scRNA samples, which completed successfully.
My command included the following options: {--solotType SmartSeq --soloStrand Unstranded --soloUMIdedup Exact NoDedup}.
I have questions on the following:
(1) In the Log.final.out file, I have ~30% of reads unmapped: too short. This percentage seems very high - Is it normal?  Is there any way I can reduce this?
(2) Regarding UMI - for SmartSeq plate-based reads which don't have UMI nor barcode, I don't understand why there is a "UMIperCellSorted.txt" file among the outputs from STAR solo.  And the Summary.csv file also contains UMI specific items such as "UMIs in cells", "Mean/Median UMI per cell".  Should/Shouldn't these UMI items be there?
(3) My STAR solo alignment was done for 2 cells (or samples) - How come in the matrix.mtx file, the rows contains 4 columns?  Is there any reference I can read up on what these columns are (and how to process this matrix.mtx with the other two .tsv files) ?

Thanks for your help xxx Amy

**Comment by alexdobin** [comment]
Hi Amy,

(1) 30% is on the high side for unmapped reads. I would start by BLASTing the unmapped reads to see if they map to some other species which would point to contamination. Another test is to map Read1 and Read2 separately to see if this improves the alignments significantly.
(2) It still uses the UMI terminology, but what it means here is "deduplicated reads" - since you used "Exact" as the first  ```--soloUMIdedup``` option.
(3) The 4 columns are
col 1 = gene index
col 2 = cell index (these should be 1 and 2 only since you have 2 cells)
col 3 = read count with start/end deduplication ("Exact")
col 4 = read count without deduplication ("NoDedup")
I would recommend switching to release 2.7.8a where the two dedup options are output in sepate matrices with 3 columns each.
Such 3 column matrices can be loaded into standard scRNA-seq analysis software such as Seurat or scanpy.

Cheers
Alex

**Comment by awlButler** [comment]
Hi Alex,
Thank you for your reply, which is very helpful.
Kind regards ... Amy

**Comment by dincaslan** [comment]
Hi, Alex and the team,

Thanks for the STAR and STARSolo. I am a huge  fan of your tools.

I am wondering whether you are planning to implement any function to demultiplex HTO/&ADT oligo barcodes to assign samples/cell surface markers. It is like the -multi option with relevant feature files in CellRanger.
 (Apologies if there is any and I am not aware yet. If not, it would be super convenient).

Many thanks,
Best regards

**Comment by alexdobin** [comment]
Hi @dincaslan 

this has been on my radar for a few years now, but I do not have bandwidth to do it in the near future, unfortunately.
